src/main/kotlin/no/nav/henvendelsesarkiv/Database.kt

```diff
@@ -0,0 +1,93 @@
+import no.nav.henvendelsesarkiv.model.Vedlegg
+import no.nav.henvendelsesarkiv.model.VedleggMapper
+import org.springframework.jdbc.core.JdbcTemplate
+
+private const val ARKIVPOST_SQL = """
+            INSERT INTO arkivpost(arkivpostId, arkivertDato, mottattDato, utgaarDato, temagruppe, arkivpostType, dokumentType,
+                                  kryssreferanseId, kanal, aktoerId, fodselsnummer, navIdent, innhold, journalfoerendeEnhet,
+                                  status, kategorikode, signert, erOrganInternt, begrensetPartInnsyn, sensitiv
+            ) VALUES (
+                ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?
+            )
+            """
+
+private const val VEDLEGG_SQL = """
+            INSERT INTO vedlegg(
+                arkivpostId, filnavn, filtype, variantformat, tittel, brevkode, strukturert, dokument
+            ) VALUES (
+                ?, ?, ?, ?, ?, ?, ?, ?
+            )
+            """
+
+class Database constructor(private val jt: JdbcTemplate, private val useHsql: Boolean = false) {
+
+    fun hentHenvendelse(id: Long): Arkivpost? {
+        val arkivpostSql = "SELECT * FROM arkivpost WHERE arkivpostId = ?"
+        val vedleggSql = "SELECT * FROM vedlegg WHERE arkivpostId = ?"
+        val arkivpost = jt.queryForObject(arkivpostSql, ArkivpostMapper(), id)
+        if(arkivpost != null) {
+            jt.query(vedleggSql, VedleggMapper(), id).forEach { arkivpost.vedleggListe.add(it) }
+        }
+        return arkivpost
+    }
+
+    fun opprettHenvendelse(arkivpost: Arkivpost): Long {
+        arkivpost.arkivpostId = nextSequenceValue()
+        jt.update(ARKIVPOST_SQL) {
+            setLong(it, 1, arkivpost.arkivpostId)
+            setTimestamp(it, 2, arkivpost.arkivertDato)
+            setTimestamp(it, 3, arkivpost.mottattDato)
+            setTimestamp(it, 4, arkivpost.utgaarDato)
+            setString(it, 5, arkivpost.temagruppe)
+            setString(it, 6, arkivpost.arkivpostType)
+            setString(it, 7, arkivpost.dokumentType)
+            setString(it, 8, arkivpost.kryssreferanseId)
+            setString(it, 9, arkivpost.kanal)
+            setString(it, 10, arkivpost.aktoerId)
+            setString(it, 11, arkivpost.fodselsnummer)
+            setString(it, 12, arkivpost.navIdent)
+            setString(it, 13, arkivpost.innhold)
+            setString(it, 14, arkivpost.journalfoerendeEnhet)
+            setString(it, 15, arkivpost.status)
+            setString(it, 16, arkivpost.kategorikode)
+            setBoolean(it, 17, arkivpost.signert)
+            setBoolean(it, 18, arkivpost.erOrganInternt)
+            setBoolean(it, 19, arkivpost.begrensetPartInnsyn)
+            setBoolean(it, 20, arkivpost.sensitiv)
+        }
+        arkivpost.vedleggListe.forEach{ opprettVedlegg(arkivpost.arkivpostId, it) }
+        return arkivpost.arkivpostId
+    }
+
+    private fun opprettVedlegg(arkivpostId: Long, vedlegg: Vedlegg) {
```

Comment:
Kaster ikke databasen exceptions ved feilsituasjoner? Bare tenkte på om vi måtte håndtere det, men kanskje det holder med bare å slippe løs en runtime-error.

Reply:
skal vurdere dette, det bør håndteres, men kanskje på toppnivå, i stedet for å sjekke overalt.